python/katana/cpp/libsupport/URI.pxd

```diff
@@ -0,0 +1,19 @@
+from libcpp cimport bool
+from libcpp.string cimport string
+
+from katana.cpp.libsupport.result cimport Result, raise_error_code
+
+cdef extern from "katana/URI.h" namespace "katana" nogil:
+
+    cdef cppclass URI:
+        @staticmethod
+        Result[URI] Make(const string& str)
+        string path() const
+        string string() const
+        bool empty() const
+
+cdef inline URI handle_result_URI(Result[URI] res) nogil except *:
+    if not res.has_value():
+        with gil:
+            raise_error_code(res.error())
+    return res.value()
```

Comment:
@arthurp, do I not need this file?  I am dealing with URIs in python code, does that mean I need a pybind wrapper?

Reply:
You do not need this file. And you shouldn't need a pybind wrapper for URI at all. Instead just convert to URI in the wrappers for functions that now take URI.